algorithms/C/stacks/stack-using-linked-list.c

```diff
@@ -0,0 +1,182 @@
+// Implementation of stack using Linked List
+
+/*
+Implement the following stack operations : 
+1. Push 
+2. Pop 
+3. Peek
+4. Display
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+
+struct Stack
+{
+    int data;
+    struct Stack *next;
+};
+
+struct Stack *top = NULL;
+
+// Push Element at the top of stack
+void push()
+{
+    int x;
+    printf("Enter Element to be pushed\n");
+    scanf("%d", &x);
+    struct Stack *newNode = (struct Stack *)malloc(sizeof(struct Stack));
```

Comment:
currently there is no way to delete this allocated memory apart from repetitively call to `pop ()` till the list is empty. 
I know implanting RAII in C can be a pain in the back side, but you could at least make the cleanup process a bit simple by proving a function to clear the entire list and give a practical way for user to avoid memory leaks as much as possible.

Reply:
I didnt get your point. Will you please elaborate ?

Reply:
@AnishLohiya I am talking about a function to erase the entire linked list ( something like a destructor in any OOP language ) so that cleaning up the list becomes relatively easier 

There can be multiple ways to implement it, one can be to simply repeatedly call `pop ()` till the list is empty.

Reply:
@Arsenic-ATG  You want me to add one more function which erases the entire linked list in the code ?

Reply:
@AnishLohiya exactly 
If you need help with it or want to do that as other contribution, then it's completely fine. Just let me know :)

Reply:
> @AnishLohiya exactly If you need help with it or want to do that as other contribution, then it's completely fine. Just let me know :)

Done 😊 @Arsenic-ATG